SETUP.md

```diff
@@ -0,0 +1,83 @@
+# Minio setup
+
+## Minio manual setup
+* Minio is used to setup local S3 servers
+* Download minio server binary from https://min.io/download
+* Make sure to add minio binary to any of the system paths (/home/<user>/bin/, /usr/local/bin/ or /bin/)
+* Create a local folder/directory that will be emulated as the buckets
+* Create a temporary directory say ```mkdir /tmp/transfer-eval/```
+* Create 2 directories inside it: ```mkdir /tmp/transfer-eval/src``` & ```mkdir /tmp/transfer-eval/dest```
+
+
+## Setup Source Buckets
+* Open the first terminal (say A)
+* Go to the emulate bucket directory (say /tmp/transfer-eval)
+* Set the username and password in this command that we will use for the authentication while transferring + run the source server (`MINIO_ROOT_USER=<username> MINIO_ROOT_PASSWORD=<password> minio server ./<path> --address :<port>`)
+* This will run the source bucket server at port <port> Eg:  ```MINIO_ROOT_USER=admin MINIO_ROOT_PASSWORD=password minio server ./src --address :8080```
+
+
+## Setup Destination Buckets
+* Open the first terminal (say B)
+* Go to the emulate bucket directory (say /tmp/transfer-eval)
+* Set the username and password in this command that we will use for the authentication while transferring + run the source server (`MINIO_ROOT_USER=<username> MINIO_ROOT_PASSWORD=<password> minio server ./<path> --address :<port>`)
+* This will run the destination bucket server at port <port> Eg:  ```MINIO_ROOT_USER=admin MINIO_ROOT_PASSWORD=password minio server ./dest --address :8090```
+
+## Emulate dummy source files
+
+* Use any UNIX tool to create a dummy file (it can be anything).
+* For brevity, we use `dd` command
+```bash
+dd if=/dev/zero of=/tmp/transfer-eval/src/testfile.1 bs=1024 count=1000000
+```
+*  This creates a dummy file ~1GB
+
+or
+
+## Minio automated setup scripts
+
+* The scripts to setup the minio source and destination servers can be found under `/scripts` folder
+* Execute `./buckets.sh` to create the buckets
+* Execute `./start-minio.sh` to start the source and destination servers
+
+### Create dummy buckets automatically
+
+`./buckets.sh` creates temporary buckets:
+- `/tmp/transfer-eval/src/`
+- `/tmp/transfer-eval/dest/`
+
+### Run minio source/destination servers directly using YAML config
+
+Check the configuration section in [README](README.md) for creating the config.
+Once we have the YAML config, we can use it to directly initialize minio servers.
```

Comment:
Does this mean we don't have to perform the steps above?

Reply:
yes
we have two ways to set up the MINIO servers i.e manual and automated way.
If the user prefers to run the automated script then no need to run the manual steps.

Reply:
As Uday said, 2 ways to perform the transfer. Manual mode can give more control over where and how to transfer.

Reply:
might need to make that a little clearer